### PR TITLE
Limit locale resource to linux platforms

### PR DESCRIPTION
**Pull request by btm**
Should fix a pipeline failure.

### CODE REVIEW

File: lib/chef/resource/locale.rb [hunk 1]
```diff
@@ -24,6 +24,7 @@ class Locale < Chef::Resource
 
       description "Use the locale resource to set the system's locale."
       introduced "14.5"
+      provides :locale, platform_family: %w{rhel fedora amazon debian}
```
Comment (lamont-granquist) [inline lib/chef/resource/locale.rb]: we've been going away from doing this.  stuff like bff_package being still declared on linux is useful for chefspec for example.

**Comment by tas50** [comment]
We need to figure out a way to do this without reverting to lwrps, but I guess this achieves the end result for now.

**Comment by lamont-granquist** [comment]
@btm you should be able to do this as a custom resource with:

```ruby
action_classs do
  def define_resource_requirements
    [....]
  end
end
```

**Comment by btm** [comment]
Replaced with a custom resource solution as suggested in: https://github.com/chef/chef/pull/8375

**Pull request closed without merging**